### PR TITLE
[14.0] ISPN-14530 Override all wrapping methods in EncoderCache

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-14530

### CODE REVIEW

**Comment by jabolina** [comment]
CI is complaining about some imports.

Maybe I missed some other method, but `getAndPutAll` was not overridden.

**Comment by tristantarrant** [comment]
I added that as well as all the `*AsyncReturnEntry` ones and `touch`

File: core/src/main/java/org/infinispan/cache/impl/EncoderCache.java [hunk 1]
```diff
@@ -972,4 +1137,9 @@ public CompletableFuture<CacheEntry<K, V>> removeAsyncEntry(Object key) {
       K keyToStorage = keyToStorage(key);
       return cache.removeAsyncEntry(keyToStorage).thenApply(e -> unwrapCacheEntry(key, keyToStorage, e));
    }
+
+   @Override
+   public CompletionStage<Boolean> touch(Object key, boolean touchEvenIfExpired) {
```
Comment (jabolina) [inline core/src/main/java/org/infinispan/cache/impl/EncoderCache.java:1142]: There is a `touch(Object, int, boolean)`, too.

Comment (tristantarrant) [reply core/src/main/java/org/infinispan/cache/impl/EncoderCache.java:1142]: Done

**Comment by wburns** [comment]
Compilation errors.

**Comment by tristantarrant** [comment]
Fixed

**Review by wburns — APPROVED** [review]
Waiting for CI

**Comment by wburns** [comment]
Integrated into 14.0, thanks @tristantarrant !

**Pull request merged**